x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfigUpdateTests.java

```diff
@@ -59,6 +60,9 @@ public static DataFrameAnalyticsConfigUpdate randomUpdate(String id) {
         if (randomBoolean()) {
             builder.setMaxNumThreads(randomIntBetween(1, 20));
         }
+        if (randomBoolean()) {
```

Comment:
Would it make sense to also add a separate unit test for merging the update? Analogous to `testMergeWithConfig_UpdatedDescription`?

Reply:
Good point. The new test reveals that merge doesn't work correctly.